time timestamp
int32 signalStrength
bool dataOk
